scripts/lint-linkcheck.mjs

```diff
@@ -0,0 +1,179 @@
+			m => m[1]
+		))];
+
+		return uniquePagePaths;
+	}
+
+	/**
+	 * Parses multiple HTML pages based on their pathnames and builds an index of their contents.
+	 */
+	parsePages (pathnames) {
+		const pages = {};
+		pathnames.forEach(pathname => {
+			pages[pathname] = this.parsePage(pathname);
+		});
+
+		return pages;
+	}
+
+	/**
+	 * Parses an HTML page based on its pathname and builds an index of its contents.
+	 */
+	parsePage (pathname) {
+		const href = this.pathnameToHref(pathname);
+		const htmlFilePath = this.pathnameToHtmlFilePath(pathname);
+
+		if (!fs.existsSync(htmlFilePath)) {
+			throw new Error('Failed to find HTML file referenced by sitemap: ' + htmlFilePath);
+		}
+
+		const dom = htmlparser2.parseDocument(fs.readFileSync(htmlFilePath));
+		const anchors = htmlparser2.DomUtils
+			.getElementsByTagName('a', dom, true);
+		
+		// Build a list of unique link hrefs on the page
+		const linkHrefs = [...new Set(anchors
+			.map(el => el.attribs.href)
+		)];
+
+		// Build a list of hashes provided by the page (mostly used as scroll targets)
+		const anchorNames = anchors
+			.map(el => el.attribs.name)
+			.filter(name => name !== undefined);
+		const ids = htmlparser2.DomUtils
+			.getElements({ id: id => id !== undefined }, dom, true)
+			.map(el => el.attribs.id);
+		const hashes = [...anchorNames, ...ids]
+			.map(name => `#${name}`);
+		
+		return {
+			pathname,
+			href,
+			htmlFilePath,
+			linkHrefs,
+			hashes,
+		};
+	}
+
+	/**
+	 * Goes through all pre-parsed and indexed pages, checks their links,
+	 * and returns an array containing all broken links (if any).
+	 */
+	findBrokenLinks (pages) {
+		var brokenLinks = [];
+
+		Object.values(pages).forEach(page => {
+			// Go through all link hrefs on the page
+			page.linkHrefs.forEach(linkHref => {
+				const url = new URL(linkHref, page.href);
+
+				// Ignore external URLs
+				if (!url.href.startsWith(this.baseUrl))
+					return;
+				
+				var linkPathname = url.pathname;
+				if (!linkPathname.endsWith('/')) {
+					linkPathname += '/';
+				}
+				const linkedPage = pages[linkPathname];
+				const isMissingPage = !linkedPage;
+
+				const decodedHash = url.hash && decodeURIComponent(url.hash);
+				const isMissingHash = (
+					!isMissingPage &&
+					(decodedHash && !linkedPage.hashes.includes(decodedHash))
+				);
+
+				if (isMissingPage || isMissingHash) {
+					brokenLinks.push({
+						page,
+						href: url.href,
+						isMissingPage,
+						isMissingHash,
+					});
+				}
+			});
+		});
+		
+		return brokenLinks;
+	}
+
+	/**
+	 * Outputs the result of the broken link check to the console.
+	 */
+	outputResult (brokenLinks) {
+		var total = brokenLinks.length;
+
+		if (total > 0) {
+			var lastPage;
+			brokenLinks.forEach(brokenLink => {
+				if (lastPage !== brokenLink.page) {
+					console.log(`\n${brokenLink.page.pathname}`);
+					lastPage = brokenLink.page;
+				}
+				console.log(`  ${brokenLink.isMissingHash ? '🔗' : '❌'} ${brokenLink.href}`);
```

Comment:
When reading the report in my terminal I didn’t understand what these emojis meant. The link icon actually read as “link” to me rather than hash. How about using a `#`?

```suggestion
				console.log(`  ${brokenLink.isMissingHash ? '#️⃣' : '❌'} ${brokenLink.href}`);
```